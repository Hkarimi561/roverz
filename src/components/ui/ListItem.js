/**
 * List Items
 *
     <ListItem title={'Hello World'} />
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ListItem } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppStyles } from '../../theme/';
import t from '../../i18n';


/* Component ==================================================================== */
class CustomListItem extends Component {
  static propTypes = {
    containerStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    titleStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    subtitleStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    avatarStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
  }

  static defaultProps = {
    containerStyle: [],
    titleStyle: [],
    subtitleStyle: [],
    avatarStyle: [],
  }

  listItemProps = () => {
    // Defaults
    let props = {
      title: t('lbl_coming_soon'),
      chevronColor: AppColors.textSecondary,
      underlayColor: AppColors.border,
      ...this.props,
      containerStyle: [{
        backgroundColor: AppColors.listItemBackground,
        paddingTop: 12,
        paddingBottom: 12,
        borderTopColor: AppColors.border,
        borderBottomColor: AppColors.border,
      }],
      titleStyle: [AppStyles.ListItemTitle],
      subtitleStyle: [AppStyles.subtext1],
      avatarStyle: [AppStyles.avatar],
    };

    props = [props, this.props.containerStyle, this.props.titleStyle, this.props.subtitleStyle, this.props.avatarStyle];
    return props;
  }

  render = () => (<ListItem {...this.listItemProps()} />);
}

/* Export Component ==================================================================== */
export default CustomListItem;
