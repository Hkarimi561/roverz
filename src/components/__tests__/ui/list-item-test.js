/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from '@ui/ListItem';

/* eslint-disable global-require */
it('ListItem renders correctly', () => {
  const tree = renderer.create(
    <ListItem
      title={'Hello world'}
      subTitle={'Sub title'}
      roundAvatar
      avatar={require('../../../image.png')}  // eslint-disable-line global-require
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
/* eslint-enable global-require */
