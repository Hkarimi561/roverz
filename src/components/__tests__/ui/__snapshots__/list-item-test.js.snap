// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<View
  0={
    Object {
      "avatar": 1,
      "avatarStyle": Array [
        Object {
          "borderRadius": 25,
          "height": 50,
          "marginTop": 5,
          "width": 50,
        },
      ],
      "chevronColor": "#777777",
      "containerStyle": Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#D0D1D5",
          "borderTopColor": "#D0D1D5",
          "paddingBottom": 12,
          "paddingTop": 12,
        },
      ],
      "roundAvatar": true,
      "subTitle": "Sub title",
      "subtitleStyle": Array [
        Object {
          "color": "#777777",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 12.6,
          "fontWeight": "500",
          "lineHeight": 18,
        },
      ],
      "title": "Hello world",
      "titleStyle": Array [
        Object {
          "color": "#222222",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 18,
          "fontWeight": "400",
          "height": 25,
          "marginBottom": 2,
        },
      ],
      "underlayColor": "#D0D1D5",
    }
  }
  1={Array []}
  2={Array []}
  3={Array []}
  4={Array []}
  onLongPress={undefined}
  onPress={undefined}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "paddingBottom": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginLeft": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={undefined}
      >
        <View />
      </View>
      <View
        style={undefined}
      >
        <View />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.15,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bdc6cf",
                "fontSize": 28,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          î—Œ
        </Text>
      </View>
    </View>
  </View>
</View>
`;
